discovery_engine/lib/src/ffi/types/embedding.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi' show Pointer, FloatPointer;
+import 'dart:typed_data' show Float32List;
+
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart'
+    show RustEmbedding1;
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+
+extension Embedding1Ffi on Float32List {
+  void writeNative(
+    final Pointer<RustEmbedding1> place,
+  ) {
+    final len = length;
+    final buffer = ffi.alloc_uninitialized_f32_slice(len);
```

Comment:
Why we can't use the same approach we used for strings?

Reply:
because `ndarray::Array` has I think no `set_len` method (I think, it would conflict with it's Dimension design).

but we could add methods for `Box<[u8]>` and then use this approach for `String`.